Mage.Sets/src/mage/cards/s/StrefanMaurerProgenitor.java

```diff
@@ -0,0 +1,215 @@
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.common.FilterCreatureCard;
+import mage.filter.predicate.permanent.TokenPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.BloodToken;
+import mage.players.Player;
+import mage.players.Players;
+import mage.target.TargetCard;
+import mage.target.common.TargetCardInHand;
+import mage.target.common.TargetControlledPermanent;
+import mage.target.targetpointer.FixedTarget;
+import mage.watchers.Watcher;
+
+import java.util.*;
+
+/**
+ * @author Alex-Vasile
+ */
+public class StrefanMaurerProgenitor extends CardImpl {
+
+    private static final Hint hint = new ValueHint("Players who lost life this turn", NumberPlayersLostLifeDynamicValue.instance);
+    private static final FilterControlledPermanent bloodTokenFilter = new FilterControlledPermanent("Blood tokens");
+
+    static {
+        bloodTokenFilter.add(SubType.BLOOD.getPredicate());
+        bloodTokenFilter.add(TokenPredicate.TRUE);
+    }
+
+    public StrefanMaurerProgenitor(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{B}{R}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.VAMPIRE);
+        this.subtype.add(SubType.NOBLE);
+
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(2);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // At the beginning of your end step, create a Blood token for each player who lost life this turn.
+        this.addAbility(
+                new BeginningOfEndStepTriggeredAbility(
+                        new CreateBloodTokenEffect(),
+                        TargetController.YOU,
+                        false)
+                        .addHint(hint),
+                new PlayersWhoLifeLostWatcher()
+        );
+
+        // Whenever Strefan attacks, you may sacrifice two Blood tokens.
+        // If you do, you may put a Vampire card from your hand onto the battlefield tapped and attacking.
+        // It gains indestructible until end of turn.
+        this.addAbility(new AttacksTriggeredAbility(
+                new DoIfCostPaid(
+                        new PlayVampireEffect(),
+                        new SacrificeTargetCost(new TargetControlledPermanent(
+                                2,
+                                2,
+                                bloodTokenFilter,
+                                true)
+                        )),
+                false,
+                "When Strefan attacks, "
+        ));
+    }
+
+    private StrefanMaurerProgenitor(final StrefanMaurerProgenitor card) { super(card);}
+
+    @Override
+    public Card copy() {
+        return new StrefanMaurerProgenitor(this);
+    }
+}
+
+class PlayVampireEffect extends OneShotEffect {
```

Comment:
I would rename this to have the card name in the class name (like `StrefanMaurerProgenitorPlayEffect` or something) just so it's clear this is only meant to be used for this card.

These classes end up being visible to every other card in the package (so everything that starts with 'S') so we don't want them getting confused for generic classes that can be used everywhere.

This goes for you dynamic value too and every other class/enum/etc you write specifically for a single card like this.  If it's meant to be generic, it should go in it's own file with the rest of the common classes.